// djsclibymech/src/assets/fileCreaters/command_creater.js
const fs = require("fs");
const path = require("path");
const symbols = require("log-symbols");
const chalk = require('chalk');
module.exports = async function (name, category, description, usage, cooldown) {
    try {
        const dirPath = path.resolve(`commands/${category}`);
        const filePath = path.resolve(`commands/${category}/${name}.js`);
        if (fs.existsSync(filePath)) return console.log(chalk.red(`${symbols.error} ${filePath} already exists, please delete that file and try again!`));

        if (fs.existsSync(dirPath)) {
            fs.writeFileSync(filePath,
                `const Discord = require('discord.js');
const config = require('../../config.json')
      
module.exports = {
name: '${name}',
category: '${category}',
description: '${description}',
usage: '${usage}',
cooldown: '${cooldown}',
    async execute(client, message, args) {
        message.channel.send('${name} Command Tested')
    }
}`);
            console.log(chalk.green(`${symbols.success} ${name}.js has been created!`))
        } else {
            await fs.mkdirSync(dirPath);
            await fs.writeFileSync(filePath,
                `const Discord = require('discord.js');
                const config = require('../../config.json')
                      
                module.exports = {
                name: '${name}',
                category: '${category}',
                description: '${description}',
                usage: '${usage}',
                cooldown: '${cooldown}',
                    async execute(client, message, args) {
                        message.channel.send('${name} Command Tested')
                    }
                }`);
            console.log(chalk.green(`${symbols.success} ${name}.js has been created!`))
        }
    } catch (err) {
        console.log(chalk.red(`${symbols.error} Command Handler has not been initialised!`)
        );
    }
}
